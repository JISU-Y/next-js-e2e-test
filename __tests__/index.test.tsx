import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Home from "@/pages"

describe("Home", () => {
  it("renders a heading", () => {
    const { container } = render(<Home />)

    const heading = screen.getByRole("heading", {
      name: /next app/i,
    })
    const justButton = screen.getByRole("button", {
      name: /just/i,
    })

    expect(container).toMatchSnapshot()
    expect(heading).toBeInTheDocument()
    expect(justButton).toBeInTheDocument()
  })

  it("should call a fn once when clicking just button", async () => {
    const mockButtonClick = jest.fn()

    render(<Home />)

    const justButton = screen.getByRole("button", {
      name: /just/i,
    })

    justButton.onclick = mockButtonClick
    await userEvent.click(justButton)

    expect(mockButtonClick).toHaveBeenCalledTimes(1)
  })
})
