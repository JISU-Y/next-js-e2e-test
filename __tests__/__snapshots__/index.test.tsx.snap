// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <main>
    <h1>
      next app
    </h1>
    <button>
      just button
    </button>
  </main>
</div>
`;
